Metrics.go

```diff
@@ -209,12 +219,26 @@ func (ctx *Metrics) addHistogramWithBuckets(subsystem, name, help string, bucket
 	return &mh
 }
 
+// RecordTimeElapsed adds the elapsed time since the specified start to the histogram in seconds and to the linked
+// summary in milliseconds.
 func (histogram *MetricsHistogram) RecordTimeElapsed(start time.Time) {
 	elapsed := float64(time.Since(start).Seconds())
 	histogram.hist.Observe(elapsed)         // The default histogram buckets are recorded in seconds
 	histogram.sum.Observe(elapsed * 1000.0) // While we have summaries in milliseconds
 }
 
+// RecordDuration adds the elapsed time since the specified start to the histogram in the specified unit of time
+// and to the linked summary in milliseconds.
+func (histogram *MetricsHistogram) RecordDuration(start time.Time, unit time.Duration) {
+	since := time.Since(start)
+	elapsedSeconds := float64(since.Seconds())
+	elapsedUnits := float64(since.Truncate(unit))
+
+	histogram.hist.Observe(elapsedUnits)
+	histogram.sum.Observe(elapsedSeconds * 1000.0)
```

Comment:
Is looks like seconds are multiplied by 1000. Is it correct?

Reply:
Unfortunately, yes. It's to convert seconds to milliseconds. Normally we would use a time constant like `time.Second`, but there's no constant for milliseconds.